src/main/java/gift/web/AdminController.java

```diff
@@ -55,7 +60,10 @@ public String editProductForm(@PathVariable Long id, Model model) {
     }
 
     @PostMapping("/edit/{id}")
-    public String editProduct(@PathVariable Long id, @ModelAttribute Product product) {
+    public String editProduct(@PathVariable Long id, @ModelAttribute @Valid Product product, BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            return "edit";
+        }
```

Comment:
질문에 답변드린대로 이렇게 수정하면 잘 동작할 겁니다!
```suggestion
    public String editProduct(@PathVariable Long id, @ModelAttribute @Valid Product product, BindingResult bindingResult, Model model) {
        if (bindingResult.hasErrors()) {
            model.addAttribute("product", product);
            model.addAttribute("org.springframework.validation.BindingResult.product", bindingResult);
            return "edit";
        }
```